docs/pages/experiments/slider-marks.tsx

```diff
@@ -0,0 +1,303 @@
+import * as React from 'react';
+import * as Slider from '@base_ui/react/Slider';
+import { useSliderContext } from '@base_ui/react/Slider';
+import { TrackFill } from './slider';
+
+const STOPS = [
+  {
+    value: 0,
+    label: '0°C',
+  },
+  {
+    value: 25,
+    label: '25°C',
+  },
+  {
+    value: 50,
+    label: '50°C',
+  },
+  {
+    value: 75,
+    label: '75°C',
+  },
+  {
+    value: 100,
+    label: '100°C',
+  },
+];
+
+function getSliderThumbAriaValueText(value: number) {
+  return `${value}°C`;
+}
+
+function MarkWithLabel(props: { index: number; value: number; label: string; inverted?: boolean }) {
+  const { index, value, label, inverted = false } = props;
+  const { isRtl, values } = useSliderContext();
+  const isFilled = inverted ? value >= values[0] : values[0] >= value;
+  return (
+    <React.Fragment>
+      <span
+        className="TempSlider-mark"
+        data-filled={isFilled}
+        data-index={index}
+        data-inverted={inverted}
+        dir={isRtl ? 'rtl' : 'ltr'}
+        style={{
+          // use percentageValues to determine this offset
+          // instead of values if min-max isn't 0-100
+          [isRtl ? 'right' : 'left']: `${value}%`,
+        }}
+      />
+      <span
+        aria-hidden="true"
+        className="TempSlider-markLabel"
+        data-filled={isFilled}
+        data-index={index}
+        data-inverted={inverted}
+        dir={isRtl ? 'rtl' : 'ltr'}
+        style={{
+          // use percentageValues to determine this offset
+          // instead of values if min-max isn't 0-100
+          [isRtl ? 'right' : 'left']: `${value}%`,
+        }}
+      >
+        {label}
+      </span>
+    </React.Fragment>
+  );
+}
+
+export default function App() {
+  return (
+    <div className="App">
+      <Slider.Root className="TempSlider" defaultValue={40}>
+        <pre>LTR</pre>
+        <Slider.Output className="TempSlider-output" />
+        <Slider.Track className="TempSlider-track">
+          {STOPS.map((mark, index) => (
+            <MarkWithLabel
+              key={`mark-${index}`}
+              index={index}
+              label={mark.label}
+              value={mark.value}
+            />
+          ))}
+          <TrackFill className="TempSlider-track-fill" />
+          <Slider.Thumb
+            className="TempSlider-thumb"
+            getAriaValueText={getSliderThumbAriaValueText}
+          />
+        </Slider.Track>
+      </Slider.Root>
+
+      <Slider.Root className="TempSlider" defaultValue={40} isRtl>
```

Comment:
The RTL and RTL inverted fill demos suffer from the same problem we had before - when a track is clicked, the thumb doesn't appear exactly under the mouse cursor but it's slightly offset.

Reply:
Fixed - thanks for spotting this, I forgot to account for the offset direction